src/nvim/ex_docmd.c

```diff
@@ -9624,26 +9624,34 @@ static void ex_folddo(exarg_T *eap)
 
 static void ex_terminal(exarg_T *eap)
 {
-  char *name = (char *)p_sh;  // Default to 'shell' if {cmd} is not given.
-  bool mustfree = false;
-  char *lquote = "['";
-  char *rquote = "']";
+  char ex_cmd[1024];
 
+  // We will call termopen() with shell as an arglist if not given a {cmd}.
   if (*eap->arg != NUL) {
-    name = (char *)vim_strsave_escaped(eap->arg, (char_u *)"\"\\");
-    mustfree = true;
-    lquote = rquote = "\"";
+    char *name = (char *)vim_strsave_escaped(eap->arg, (char_u *)"\"\\");
+    snprintf(ex_cmd, sizeof(ex_cmd),
+             ":enew%s | call termopen(\"%s\") | startinsert",
+             eap->forceit ? "!" : "", name);
+    xfree(name);
+  } else {
+    char **argv = build_argv((char *)p_sh);
```

Comment:
Yes, I do believe I tried that.

I have tried adding it back now, without the early validation check and return, it looks like there is a hung build. Which might be because the assertion on `src/nvim/shell.c:71` would fail if shell was blank and cmd/extra args are both NULL.

Adding the early EMSG to avoid this seems to also cause test failures that could be due to early returning of `ex_terminal`. I'm going to revert back to using `build_argv`.